### ISSUE TITLE
sys-net still starts automatically after deselecting "Start VM automatically on boot"

### ISSUE THREAD

**Issue by andrewdavidwong** [QubesOS/qubes-issues#3606]
#### Qubes OS version:

`R3.2`

---

### Steps to reproduce the behavior:

1. In Qubes Manager, open the VM settings for `sys-net`.
2. Deselect "Start VM automatically on boot".
3. Click "OK".
4. Reboot the whole system.

### Expected behavior:

`sys-net` no longer starts automatically on boot.

### Actual behavior:

`sys-net` still starts automatically on boot.

### General notes:

In order to stop `sys-net` from starting automatically on boot, the user has to enter this command:

```
systemctl disable qubes-netvm.service
```

This is exactly what deselecting "Start VM automatically on boot" should do.

---

#### Related issues:

#3554

**Comment by donob4n** [comment]
AFAIK, qubes-netvm.service starts the default-netvm and qubes-vm@.service's (auto started VM's?) wait for it. Maybe there are some purposes for qubes-netvm that I don't find.

I think there are three options:
1) Since qubes-vm@.service relies on qvm-start, all netVM dependencies should start fine without the need of systemd waiting, so probably just removing qubes-netvm.service seems fine (qubes-reload-firewall also waits but it could be removed since this service won't be loaded without the VM already running)

2) Modify qubesmanager for detect when the vm is the default netVM and ask user if wants to change default netVM or disable qubes-netvm.service

3) Just not consider this a bug.

What do you think?

**Comment by andrewdavidwong** [comment]
> Duplicate of #3554

Did you forget our conversation about this? https://github.com/QubesOS/qubes-issues/issues/3554#issuecomment-366589413

**Comment by andrewdavidwong** [comment]
Reopening becuase this is **not** a duplicate of #3554. It's obviously a bug that the option to deselect "Start VM automatically on boot" exists for `sys-net` when deselecting this option doesn't stop `sys-net` from starting automatically on boot.

**Comment by marmarek** [comment]
qubes-netvm is bound to default netvm, not sys-net specifically. Changing property of what's _currently_ default netvm, shouldn't affect qubes-netvm service.
Even more - default netvm is actually sys-firewall, so there is no reason why changing anything in sys-net should touch qubes-netvm service.

If anything, checkbox "Start VM automatically on boot" should have some tooltip that VM autostart may be affected by qubes-netvm service, or being a netvm for another VM that is select for autostart.

**Comment by marmarek** [comment]
Both this and #3554 is about lack of information that qubes-netvm service affect starting default netvm (the difference is that initially #3554 was about sys-whonix being that default netvm, but this one about sys-net/sys-firewall). But the issue is the same.

**Comment by andrewdavidwong** [comment]
> qubes-netvm is bound to default netvm, not sys-net specifically. Changing property of what's currently default netvm, shouldn't affect qubes-netvm service.
Even more - default netvm is actually sys-firewall, so there is no reason why changing anything in sys-net should touch qubes-netvm service.

But none of this changes the fact that the bug described in this issue is a bug (the presence of a checkbox that doesn't do what it says it does).

> If anything, checkbox "Start VM automatically on boot" should have some tooltip that VM autostart may be affected by qubes-netvm service, or being a netvm for another VM that is select for autostart.

Sure, that's one potential solution, though one that would be difficult for users to understand. Other possible responses:

- Remove the checkbox
- Change the text next to the checkbox to be more accurate
- File as `wontfix` if you think it's worth the dev time this late in 3.2's lifecycle

> Both this and #3554 is about lack of information that qubes-netvm service affect starting default netvm (the difference is that initially #3554 was about sys-whonix being that default netvm, but this one about sys-net/sys-firewall). But the issue is the same.

In that case, I disagree that it's working as intended (or, if it really is, then I disagree with your intention). It is simply wrong to provide the user with a checkbox that does not do what it says it does.

**Comment by marmarek** [comment]
But do you agree those are the same issue?

**Comment by andrewdavidwong** [comment]
> But do you agree those are the same issue?

Yes, I grant that it turned out to be the same underlying problem (and, therefore, it now makes sense to have just a single issue in the tracker for this). But, in that case, #3554 should not be closed as working as intended.

If you like, you can think of it as a UX issue (personally, I think this would be too charitable): Presenting the user with a checkbox that says, "Start VM automatically on boot," where *un*checking that box results in the VM still starting automatically on boot, makes for a bad user experience, because the user's intention is being flouted. In general, providing the user with a checkbox that does nothing is problematic.
